frontend/src/hooks/useMenuActions.tsx

```diff
@@ -0,0 +1,87 @@
+import { Dispatch, SetStateAction, useCallback } from "react";
+import { fabric } from "fabric";
+import {
+  addLine,
+  addRectangle,
+  addCircle,
+  handleAddText,
+  handleGroupSelected,
+  handleRemoveSelected,
+  handleLoadFromJSON,
+} from "@/lib/fabricCanvasUtils";
+
+const useMenuActions = (
+  canvas: fabric.Canvas | null,
+  color: string,
+  size: number,
+  saveState: () => void,
+  setIsDrawingMode: Dispatch<SetStateAction<boolean>>
+) => {
+  const defaultFontSize = 20;
+
+  const performAction = useCallback(
+    (name: string) => {
```

Comment:
This can be turned into a type I guess. Something like ```type CanvasAction = 'add-line' | 'add-rectangle' | ...```. It's not strictly needed but would improve readability. We would know what the `performAction` function actually expects.

Reply:
yes, i will do it